pkg/agent/agent_readiness.go

```diff
@@ -53,6 +54,33 @@ func AllReachedGoalState(sts appsv1.StatefulSet, podGetter pod.Getter, desiredMe
 	return true, nil
 }
 
+func GetAllDesiredMembersPodState(sts appsv1.StatefulSet, podGetter pod.Getter, desiredMemberCount, targetConfigVersion int, log *zap.SugaredLogger) ([]PodState, error) {
+	var podStates []PodState
```

Comment:
nit: some optimization, since we know that the size of this slice would be we can pre-allocate here instead of appending.

Reply:
Done.